### ISSUE TITLE
'hoistTry' rule has an issue in strings

### ISSUE THREAD

**Issue by foscomputerservices** [nicklockwood/SwiftFormat#1403]
### Issue

There seems to be an issue when hoisting try out of expressions `embedded` in strings.  Take the following code for example:

```swift
public struct TestHoistTry {
    public init() throws {
        print("Hello \(try Foo.tryMe())")
    }
}

struct Foo {
    static func tryMe() throws -> String { ", World!" }
}
```

After running swiftlint we get the following:

```swift
public struct TestHoistTry {
    public init() throws {
        print("Hello \try (Foo.tryMe())") // <- try was lifted to the \
    }
}

enum Foo {
    static func tryMe() throws -> String { ", World!" }
}
```

### Environment

% swiftformat --version
0.51.2

% swift --version
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0

% xcodebuild -version
Xcode 14.2
Build version 14C18

**Comment by nicklockwood** [comment]
This has already been reported and will be fixed in the next release. Thanks!

**Comment by nicklockwood** [comment]
@foscomputerservices fixed in 0.51.3
